Sources/MapboxNavigation/SpeedLimitView.swift

```diff
@@ -106,15 +116,22 @@ public class SpeedLimitView: UIView {
     override public func draw(_ rect: CGRect) {
         super.draw(rect)
         
-        guard let speedLimit = speedLimit, let signStandard = signStandard else {
+        guard let signStandard = signStandard else {
             return
         }
         
         let formattedSpeedLimit: String
-        if speedLimit.value.isInfinite {
-            formattedSpeedLimit = "∞"
+
+        if let speedLimit = speedLimit {
+            if speedLimit.value.isInfinite {
+                formattedSpeedLimit = "∞"
+            } else {
+                formattedSpeedLimit = measurementFormatter.numberFormatter.string(for: speedLimit.value) ?? "\(speedLimit.value)"
+            }
+        } else if shouldShowUknownSpeedLimit {
+            formattedSpeedLimit = "--"
         } else {
-            formattedSpeedLimit = measurementFormatter.numberFormatter.string(for: speedLimit.value) ?? "\(speedLimit.value)"
+            return
```

Comment:
have you played around to see if this return statement gets accessed and the resulting behavior if it does?

Reply:
Yeah, I tested everything, this return statement will be hit if `speedLimit` is nil and `shouldShowUnknownSpeedLimit` is false, if this happens the view will be hidden and we don't want to draw the contents. So the old behavior is fully saved.